### ISSUE TITLE
remove auto-correct when full command/text-to-correct is typed

### ISSUE THREAD

**Issue by MaxLeiter** [thelounge/lounge#1182]
if you dont use auto-correct and just type out a command, like `/part`, you need to hit enter twice. Auto-correct should notice when you type a full command

**Comment by YaManicKill** [comment]
So, valid point but 1 thing to note is that if we have a command that is a subset of another command (for example, /ban and /banlist, you aren't necessarily finished when you type "/ban").

I know lots of other apps do this by not autocompleting when you type enter, but sending the message. Autocompleting with <tab>. But I don't really like that.

**Comment by MaxLeiter** [comment]
or just if there's only one possible autocorrect?

**Comment by YaManicKill** [comment]
Yeah, I'd definitely be happy with that. If 1 autocorrect and it is fully typed, finish autocompleting.

**Comment by realies** [comment]
It would make more sense if autocorrect is not triggered by pressing enter but by going up once (or more) with the arrow up, highlighting the correction and pressing enter. On multiple occasions, I have had to go back and manually retype the command or channel that was autocorrected.

**Comment by xPaw** [comment]
This is mildly related to #2100 bug. We could fix both of these by inserting a space after completing commands and channels. Would that be fine?

**Comment by astorije** [comment]
For commands, yes because there is no reason not to have a space after that. Not after channels for the same reason that we don't add a space after nicks.
